listenbrainz/webserver/static/js/src/Bar.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from "react";
+import { ResponsiveBar, LabelFormatter } from "@nivo/bar";
+
+export type BarProps = {
+  data: Array<object>;
+  maxValue: number;
+};
+
+export type BarState = {
+  marginLeft: number;
+};
+
+export default class Bar extends React.Component<BarProps, BarState> {
+  constructor(props: BarProps) {
+    super(props);
+
+    this.state = {
+      marginLeft: window.innerWidth / 5,
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener("resize", this.handleResize);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.handleResize);
+  }
+
+  handleResize = () => {
+    this.setState({
+      marginLeft: window.innerWidth / 5,
+    });
+  };
+
+  render() {
+    const { data, maxValue } = this.props;
+    const { marginLeft } = this.state;
+
+    const leftAlignedTick = (tick: any) => {
+      let { value } = tick;
+      if (value.length > marginLeft / 10) {
+        value = `${value.slice(0, marginLeft / 10)}...`;
+      }
+
+      return (
+        <g transform={`translate(${tick.x - marginLeft}, ${tick.y})`}>
+          <text textAnchor="start" dominantBaseline="middle">
+            {value}
+          </text>
+        </g>
+      );
+    };
+
+    const labelFormatter = (((label: string) => {
+      return (
+        <tspan x={5} textAnchor="start">
+          {label}
+        </tspan>
+      );
+    }) as unknown) as LabelFormatter;
```

Comment:
unknown?

Reply:
The definition of `LabelFormatter` expects a function which returns a `string` or a `number`, even though `JSX` elements work. I will file an issue in `nivo` regarding this but until then we will have to keep it as it is.